docs/glossary.md

```diff
@@ -64,6 +61,26 @@ users. It can be deployed in the cloud, or on your own hardware.
+
+MyST Document Engine
+: A MyST Document Engine builds {term}`MyST AST` according to the {term}`MyST Specification`[^myst]. For example, a MyST Document Engine might know how to transform a text file written in {term}`MyST Markdown` into JSON output that follows the {term}`MyST Specification`. The official MyST Document Engine is documented at [mystmd.org/guide](https://mystmd.org/guide), and is maintained by the [Jupyter Book team](https://compass.jupyterbook.org)
+
+[^myst]: Often we focus upon a MyST Engine's ability to parse {term}`MyST Markdown`. However, a MyST Engine may consume _any_ kind of input markup as long as the _result_ is a {term}`MyST AST` that follows the {term}`MyST Specification`. This is why the [official MyST Document Engine](https://mystmd.org) can parse other kinds of markup, such as a subset of LaTeX.
+
+MyST Specification
+: A specification that describes the structure and function of the MysT AST. It provides a framework for defining all of the kinds of content and metadata that a MyST document can contain. For example, the MyST specification defines how sections can contain paragraphs, how paragraphs can contain sentences, and how sentences can contain a "bolded" chunk of text. The specification is [defined at mystmd.org/spec](xref:spec/#overview).
+
+MyST AST
+: A structured representation of a MyST Document, typically built from a markup language (like MyST Markdown) by a {term}`MyST Document Engine`, that adheres to the {term}`MyST Specification`. Usually, a MyST [Abstract Syntax Tree (AST)](wiki:Abstract_syntax_tree) is represented as a JSON data structure, with metadata attached to each piece of content that describes its role, relationships with other content, etc. The AST allows us to separate the steps of parsing content (such as {term}`MyST Markdown`) from generating output like HTML, PDF, docx, etc.
+
+MyST Renderer
+: An application that consumes {term}`MyST AST` and uses it to present the contents in a particular format. For example, the MyST Document Engine has a few built-in renderers. See [the HTML renderer](https://github.com/jupyter-book/mystmd/tree/840265e0c055d29b5d5ce80f391ccb1efba7a87e/packages/myst-to-html) for an example that converts {term}`MyST AST` to HTML.
+
+MyST Theme
+: A template for conerting {term}`MyST AST` into structured output using a {term}`MyST Renderer`. For example, a theme provides the structure and rules by which you could convert {term}`MyST AST` to different kinda of Latex output based on a number of different journal templates. It could also be a way of converting {term}`MyST AST` into different kinds of HTML for various kinds of websites. The [MyST Templates GitHub Organization](https://github.com/myst-templates) is a community-maintained resource for many kinds of themes / templates.
```

Comment:
```suggestion
: A template for converting {term}`MyST AST` into structured output using a {term}`MyST Renderer`. For example, a theme provides the structure and rules by which you could convert {term}`MyST AST` to different kinds of LaTeX output based on a number of different journal templates. It could also be a way of converting {term}`MyST AST` into different kinds of HTML for various kinds of websites. The [MyST Templates GitHub Organization](https://github.com/myst-templates) is a community-maintained resource for many kinds of themes / templates.
```